File: client/Mitou/Assets/Scripts/Network/ConnectionState.cs
namespace ERA.Network
{
    /// <summary>
    /// WebSocket接続状態を表す列挙型。
    /// </summary>
    public enum ConnectionState
    {
        /// <summary>切断状態（初期状態）。</summary>
        Disconnected,

        /// <summary>接続中。</summary>
        Connecting,

        /// <summary>接続完了。</summary>
        Connected,

        /// <summary>再接続中。</summary>
        Reconnecting,

        /// <summary>エラー状態（再接続上限超過など）。</summary>
        Error
    }
}

File: client/android/app/src/main/java/com/era/android/data/websocket/ConnectionState.kt
package com.era.android.data.websocket

/**
 * WebSocket接続状態を表すsealed class
 */
sealed class ConnectionState {
    /**
     * 切断状態
     */
    object Disconnected : ConnectionState()

    /**
     * 接続試行中
     */
    object Connecting : ConnectionState()

    /**
     * 接続済み
     */
    object Connected : ConnectionState()

    /**
     * 再接続試行中
     * @property attempt 現在の試行回数（1から開始）
     * @property nextRetryMs 次回再接続までの待機時間（ミリ秒）
     */
    data class Reconnecting(val attempt: Int, val nextRetryMs: Long) : ConnectionState()

    /**
     * エラー状態
     * @property message エラーメッセージ
     * @property code エラーコード（オプション）
     */
    data class Error(val message: String, val code: Int? = null) : ConnectionState()
}
